lib/coverband/at_exit.rb

```diff
@@ -24,7 +24,10 @@ def self.register
           # * Perhaps detect heroku deployment ENV var opposed to tasks?
           #####
           default_heroku_tasks = ['assets:clean', 'assets:precompile']
-          if defined?(Rake) && Rake.respond_to?(:application) && (Rake&.application&.top_level_tasks || []).any? { |task| default_heroku_tasks.include?(task) }
+          if !Coverband.configuration.report_on_exit ||
+             (defined?(Rake) &&
+             Rake.respond_to?(:application) &&
+             (Rake&.application&.top_level_tasks || []).any? { |task| default_heroku_tasks.include?(task) })
```

Comment:
Since we are setting the flag within the environment task, are these rake tasks still necessary? I believe all of the heroku tasks depend on the environment rake task.

Reply:
so this isn't the default environment just a method called `environment` in the Coverband rake task. I think we could hook into the rake tasks environment used by things on Heroku but if we did the issue would be how can we report on rake tasks used as part of cron jobs, but not report on preloading tasks like `assets`. The issue this prevented was on deploy having the asset job ending up putting a bunch of data in runtime not load. We could likely revisit as we learned rake skips eager loading but I think that is still different than our internal rake tasks which we wan to do nothing at all.

Reply:
Great point about cron job rake tasks. I think we should leave this for now. That being said, it's tough to remember what specific data was being added to `runtime` coverage and why that was a problem. Now that we have forked tests (#251), we should probably look to add a test that exposes the need for this code. For another PR though.

Reply:
yeah agreed we need to definitely simplify and some of this underlying logic has changed I added a comment on this PR as it was a pretty fundamental change to how all this stuff works.

https://github.com/danmayer/coverband/pull/238

 Perhaps opposed to hard coded Heroku rake tasks we could move to a config `config.rake_tasks_to_ignore` similar to how we treat ignoring files, and it could have some nice defaults like the heroku rake tasks.